public/docs/_examples/server-communication/dart/lib/wiki/wiki_component.dart

```diff
@@ -0,0 +1,26 @@
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'wikipedia_service.dart';
+import 'dart:async';
+
+@Component(
+    selector: 'my-wiki',
+    template: '''
+      <h1>Wikipedia Demo</h1>
+      <p><i>Fetches after each keystroke</i></p>
+      <input #term (keyup)="search(term.value)"/>
+      <ul>
+        <li *ngFor="#item of items">{{item}}</li>
+      </ul>
+    ''',
+    providers: const [WikipediaService])
+class WikiComponent {
+  WikipediaService _wikipediaService;
+  List items;
```

Comment:
Prefer initializing items as in general you might get unexpected behaviour as things will be null for a brief moment.